src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -62,6 +70,53 @@ class TasksController(
     }
   }
 
+  private def deleteTasks()(implicit identity: Identity): Route = {
+    def doKill(force: Boolean, scale: Boolean, wipe: Boolean, tasksIdToAppId: Map[Id, PathId]) = {
+      def isAuthorized(tasksIdToAppId: Map[Id, PathId]): Boolean = tasksIdToAppId.values.exists(id => !authorizer.isAuthorized(identity, UpdateRunSpec, id))
+
+      // TODO authorization check is performed twice - in the taskKiller and also here, this one must remain
+      if (isAuthorized(tasksIdToAppId)) {
+        reject(AuthDirectives.NotAuthorized(HttpPluginFacade.response(authorizer.handleNotAuthorized(identity, _))))
+      } else {
+        val maybeInstances: Future[Iterable[Option[Instance]]] = Future.sequence(tasksIdToAppId.view
+          .map { case (taskId, _) => instanceTracker.instancesBySpec.map(_.instance(taskId)) })
+        val tasksByAppId: Future[Map[PathId, Seq[Instance]]] = maybeInstances.map(i => i.flatten
+          .groupBy(instance => instance.instanceId.runSpecId)
+          .map { case (appId, instances) => appId -> instances.to[Seq] }(collection.breakOut))
+
+        if (scale) {
+          onSuccess(killAndScale(tasksByAppId, force)) { deploymentResult =>
+            complete((StatusCodes.OK, List(Headers.`Marathon-Deployment-Id`(deploymentResult.deploymentId)), deploymentResult))
+          }
+        } else {
+          onSuccess(doKillTasks(tasksByAppId, tasksIdToAppId, wipe)) { tasks =>
+            complete(TasksList(tasks).toRaml)
+          }
+        }
+      }
+    }
+
+    (entity(as[TasksToDelete])
+      & parameter('force.as[Boolean].?(false))
+      & parameter('scale.as[Boolean].?(false))
+      & parameter('wipe.as[Boolean].?(false))) { (taskIds, force, scale, wipe) =>
+        if (scale && wipe) {
+          reject(MalformedQueryParamRejection("scale, wipe", "You cannot use scale and wipe at the same time."))
+        } else {
+          val maybeTasksIdToAppId: Try[Map[Id, PathId]] = Try(taskIds.ids.map { id =>
```

Comment:
Out of curiosity, what is the benefit of `Try` here?

Reply:
since I am not mapping/transforming the Try in any way, the benefit is probably not that big. I just generally use Try object instead of oldschool try/catch everywhere, I prefer this "more functional style". If it does not conform to the coding style of the rest of the codebase, just tell me and I'll change it